### PR TITLE
Meta: Clean up webpack config file

### PR DESCRIPTION
**Pull request by fregante**
This extracts code and checks unrelated to the build to its own file.

Ideally, separately, the actual checks would be completely separate from webpack. They'd require this same file, but they wouldn't throw errors.

### CODE REVIEW

**Comment by yakov116** [comment]
@fregante can you update webpack too? I tried updating and I was unable to build

File: package.json [hunk 1]
```diff
@@ -120,7 +120,7 @@
 		"type-fest": "^0.21.3",
 		"typed-query-selector": "^2.4.1",
 		"typescript": "^4.2.3",
-		"webpack": "^5.27.1",
+		"webpack": "^5.30.0",
 		"webpack-cli": "^4.5.0",
```
Comment (yakov116) [inline package.json:124]: this too

**Comment by fregante** [comment]
It was

File: build/readme-parser.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
+/// <reference types="../source/globals" />
+import path from 'path';
+import {readFileSync} from 'fs';
+import regexJoin from 'regex-join';
+import {parse as parseMarkdown} from 'markdown-wasm/dist/markdown.node.js';
```
Comment (yakov116) [inline build/readme-parser.ts:5]:
```suggestion
import path from 'path';
import regexJoin from 'regex-join';
import {readFileSync} from 'fs';
import {parse as parseMarkdown} from 'markdown-wasm/dist/markdown.node.js';
```

File: build/readme-parser.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+	const markdownDescription = lineMatch[1].replace(/\[(.+?)]\((.+?)\)/g, urlExtracter);
+	return {
+		id,
+		description: parseMarkdown(markdownDescription),
+		screenshot: urls.find(url => /\.(png|gif)$/i.test(url))
+	};
+}
+
+function searchInHighlights(readmeContent: string, id: FeatureID): FeatureMeta | void {
+	// Feature might be highlighted in the readme
```
Comment (fregante) [inline build/readme-parser.ts:31]:
```suggestion
```

File: webpack.config.ts [hunk 4]
```diff
@@ -1,57 +1,18 @@
 /// <reference types="./source/globals" />
 
 import path from 'path';
-import {readFileSync} from 'fs';
```
Comment (yakov116) [inline webpack.config.ts:4]:
```suggestion
```

Comment (fregante) [reply webpack.config.ts]: I think actually built-in modules need to be spaced

File: build/readme-parser.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
+/// <reference types="../source/globals" />
```
Comment (yakov116) [inline build/readme-parser.ts:1]: Unrelated question. Why is this needed?

**Comment by yakov116** [comment]
Does build need to be added here?

https://github.com/sindresorhus/refined-github/blob/a34191037be2ce5ad44ca6cc7015f71c8c8a7b19/tsconfig.json#L11-L14

If yes `"octicon-svg-loader.ts"` can be removed

File: build/readme-parser.ts (hunk 5 above)
Comment (fregante) [reply build/readme-parser.ts:1]: Don't ask me!

I think I tried moving `globals` to the root and use it there but it didn't work…

**Review by yakov116 — APPROVED** [review]
Untested but looks good.

Want to regenerate the lock?

File: webpack.config.ts [hunk 6]
```diff
@@ -160,9 +106,7 @@ const config: Configuration = {
...
 };
 
-const webpackSetup = (_: string, options: webpack.WebpackOptionsNormalized): Configuration => {
+export default function webpackSetup(_: string, options: webpack.WebpackOptionsNormalized): Configuration {
 	isWatching = Boolean(options.watch);
 	return config;
-};
-
-export default webpackSetup;
+}
```
Comment (yakov116) [inline webpack.config.ts:112]: hey why no `;`?

**Comment by fregante** [comment]
It’s ok for now since only select dev dependencies have been updated.

**Comment by yakov116** [comment]
> It’s ok for now since only select dev dependencies have been updated, we don’t have to do it every time.

I got burned from this so many times. I wish I would know the reason, but since we switched to npm 15 every time I do an update if I don't delete and start over, 💥.

Probably has to do with my machine. 🤷

**Comment by fregante** [comment]
Yeah it does happen but I supposed that if it works on CI it works for everyone. Usually they're peer dependency errors where babel stuff got updated partially and it's not fully compatible across versions.

**Comment by fregante** [comment]
Tested both build and watch

**Comment by yakov116** [comment]
Me too :)

Just a heads up I really learned from this PR. How you put the code together so simply and understandable.

I look at the code and I think to myself "ya maybe one day..."

**Comment by fregante** [comment]
> When do you sleep?!

I recently moved across 12 timezones. I might have normalized my sleep schedule for now 😂

**Comment by yakov116** [comment]
I was wondering!
